### PR TITLE
[UI] Split editor component

### PR DESCRIPTION
**Pull request by mykalmax**
- [x]  Split Editor component 
- [x] Rearrange shared state between Editor and FileTree components

### CODE REVIEW

File: web/client/src/library/components/editor/EditorInspector.tsx [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+      </p>
+    </div>
+  )
+}
+
+function ModelName({ modelName }: { modelName: string }): JSX.Element {
+  return (
+    <div className="text-sm font-bold whitespace-nowrap">
+      <h3 className="ml-2">Model Name</h3>
+      <p className="mt-1 px-2 py-1 bg-secondary-10 text-secondary-500 dark:text-primary-500 dark:bg-primary-10  text-xs rounded">
```
Comment (vchan) [inline web/client/src/library/components/editor/EditorInspector.tsx:352]:
```suggestion
      <p className="mt-1 px-2 py-1 bg-secondary-10 text-secondary-500 dark:text-primary-500 dark:bg-primary-10 text-xs rounded">
```

File: web/client/src/library/components/editor/extensions/SqlMeshDialect.ts [hunk 2]
```diff
@@ -8,132 +8,138 @@ import { LanguageSupport } from '@codemirror/language'
...
-        autocomplete: completeFromList([
-          {
-            label: 'model',
-            type: 'keyword',
-            apply: 'MODEL (\n\r)',
+      cache.set('message', handler)
+
+      sqlglotWorker.addEventListener('message', handler)
+
+      console.log({ tree })
```
Comment (vchan) [inline web/client/src/library/components/editor/extensions/SqlMeshDialect.ts:77]: Debug statement?

**Review by vchan — APPROVED** [review]

File: web/client/src/context/editor.ts [hunk 3]
```diff
@@ -1,25 +1,142 @@
 import { create } from 'zustand'
+import useLocalStorage from '~/hooks/useLocalStorage'
+import { ModelFile } from '~/models'
+import { isTrue } from '~/utils'
+import { sqlglotWorker } from '~/workers'
+
+export interface Dialect {
+  dialect_title: string
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/context/editor.ts:8]: that's how we're getting it from sqlglot file

File: web/client/src/library/components/editor/EditorPreview.tsx [hunk 4]
```diff
@@ -6,28 +6,33 @@ import {
   getCoreRowModel,
   useReactTable,
 } from '@tanstack/react-table'
-import { useStoreEditor } from '../../../context/editor'
-import { useStoreFileTree } from '../../../context/fileTree'
 import { isNil, isTrue } from '~/utils'
+import { useStoreEditor } from '~/context/editor'
 
+export const EnumEditorTabs = {
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/library/components/editor/EditorPreview.tsx:12]: fixed

File: web/client/src/context/context.ts [hunk 5]
```diff
@@ -41,21 +45,29 @@ export const useStoreContext = create<ContextStore>((set, get) => ({
   environments,
   initialStartDate: undefined,
   initialEndDate: undefined,
-  getNextEnvironment(): ModelEnvironment {
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/context/context.ts:44]: we are declaring types in function signature in Interface above

**Pull request merged**